libimage/runtime.go

```diff
@@ -316,19 +316,36 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 		return img, name, err
 	}
 
-	return r.lookupImageInDigestsAndRepoTags(name, options)
+	return r.lookupImageInRepoTags(name, namedName, options)
 }
 
 // lookupImageInLocalStorage looks up the specified candidate for name in the
 // storage and checks whether it's matching the system context.
 func (r *Runtime) lookupImageInLocalStorage(name, candidate string, options *LookupImageOptions) (*Image, error) {
 	logrus.Debugf("Trying %q ...", candidate)
+
+	// First, try store.Image() which will properly work on IDs (but not on
+	// digests).
 	img, err := r.store.Image(candidate)
 	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
 		return nil, err
 	}
 	if img == nil {
```

Comment:
I think this really does not need to be heuristically parsing the input two ways.

Every single caller _knows_ which code path is relevant: it is either submitting some kind of `reference.Named`, or a string known to be an ID, or a string presumed to be a short ID.

I’d be happier with not throwing away that information and not introducing an unnecessary parsing ambiguity.

---

I.e. factor the code dealing with manifests lists (and platforms?) into a separate function, `finalizeImageLookupInLocalStorage` or something like that, and refer to it from a two separate callers, `lookupImageInLocalStorageByNamedReference` and `lookupImageInLocalStorageByIDOrPrefix`.

Then the “Do not perform any further platform checks” can be moved into only one of the two.